app/routers/analyze.py

```
@@ -0,0 +1,42 @@
+
+    # Cleanup temp file
+    os.remove(temp_file_path)
+
+    if "❌" in extracted_text:
+        raise HTTPException(status_code=500, detail="❌ Failed to extract text.")
+
+    # ✅ **Fix: Catch AI Errors**
+    try:
+        ai_response = AIClient.analyze_risk(extracted_text)
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=f"❌ AI Analysis Failed: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)